packages/siopv2-oid4vp-op-auth/src/session/OpSession.ts

```diff
@@ -302,16 +306,15 @@ export class OpSession {
     const verification: Verification = {
       presentationVerificationCallback: this.createPresentationVerificationCallback(this.context),
     }
-
     const request = await this.getAuthorizationRequest()
     const hasDefinitions = await this.hasPresentationDefinitions()
     if (hasDefinitions) {
       const totalInputDescriptors = request.presentationDefinitions?.reduce((sum, pd) => {
         return sum + pd.definition.input_descriptors.length
       }, 0)
       const totalVCs = args.verifiablePresentations?.reduce((sum, vp) => {
-        const uvp = CredentialMapper.toUniformPresentation(vp)
-        return sum + (uvp.verifiableCredential?.length ?? 0)
+        const uvp = CredentialMapper.toUniformPresentation(vp, { hasher: args.hasher ?? this.options.hasher })
+        return sum + ((uvp.verifiableCredential?.length ?? CredentialMapper.isSdJwtDecodedCredential(uvp)) ? 1 : 0)
```

Comment:
i feel like something is not right here. 1 this is almost unreadable. 2 is this looking for credentials in a presentation and then when it cannot find any we are like ok, maybe it was a sd-jwt?

this logic is kinda weird?